File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ContainerService.java
package com.sequenceiq.cloudbreak.service.cluster;

import java.util.Set;

import javax.inject.Inject;

import org.springframework.stereotype.Service;

import com.sequenceiq.cloudbreak.domain.Container;
import com.sequenceiq.cloudbreak.repository.ContainerRepository;

@Service
public class ContainerService {

    @Inject
    private ContainerRepository containerRepository;

    public Iterable<Container> save(Iterable<Container> containers) {
        return containerRepository.saveAll(containers);
    }

    public Set<Container> findContainersInCluster(Long clusterId) {
        return containerRepository.findContainersInCluster(clusterId);
    }

    public void deleteAll(Iterable<Container> containers) {
        containerRepository.deleteAll(containers);
    }

}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ContainerService.kt
package com.sequenceiq.cloudbreak.service.cluster

import com.sequenceiq.cloudbreak.domain.Container
import com.sequenceiq.cloudbreak.repository.ContainerRepository
import org.springframework.stereotype.Service

import javax.inject.Inject
import javax.transaction.Transactional

@Service
@Transactional
class ContainerService {

    @Inject
    private val containerRepository: ContainerRepository? = null

    @Transactional(javax.transaction.Transactional.TxType.NEVER)
    fun save(containers: List<Container>): Iterable<Container> {
        return containerRepository!!.save(containers)
    }

    @Transactional(javax.transaction.Transactional.TxType.NEVER)
    fun findContainersInCluster(clusterId: Long?): Set<Container> {
        return containerRepository!!.findContainersInCluster(clusterId)
    }
}
